src/metabase/util/malli/humanize.cljc

```diff
@@ -0,0 +1,71 @@
+(ns metabase.util.malli.humanize
+  (:require
+   [malli.error :as me]))
+
+(defn- resolve-error
+  "This is the same behavior as what [[malli.error/humanize]] does to resolve errors."
+  [explanation error]
+  (me/-resolve-direct-error explanation error {:wrap :message, :resolve me/-resolve-direct-error}))
+
+(defn- flatten-error
+  "Given a `[path message]` pair like
+
+    [[2] \"some error\"]
+
+  return a flattened error message like
+
+    [nil nil \"some error\"]"
+  [[path message]]
+  (if (empty? path)
+    message
+    (recur
+     [(butlast path)
+      (if (integer? (last path))
+        (me/-push [] (last path) message nil)
+        {(last path) message})])))
+
+(defn- merge-errors
+  "Merge two flattened errors into a single error, e.g.
+
+    (merge-errors {:x \"oops\"}
+                  {:x \"oh no\"})
+    ;; => {:x (\"oops\" \"oh no\")}
+
+  List-like structures are used to differentiate multiple errors (e.g., the result of an `:or` schema) from single
+  errors (which use a vector instead)."
+  [msg-1 msg-2]
+  (cond
+    (= msg-1 msg-2)
+    msg-1
+
+    (nil? msg-1)
+    msg-2
```

Comment:
Isn't
```
    (nil? msg-2)
    msg-1
```
missing? The docstring suggests that the arguments should be handled symmetrically.